health/service.go

```diff
@@ -0,0 +1,151 @@
+package health
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+)
+
+// CheckPath is the baked in SOH endpoint path.
+const CheckPath = "/soh"
+
+// Service provides a mechanism to update a service SOH status.
+type Service struct {
+	mu sync.Mutex
+
+	// status is used to indicate whether the service is running
+	status bool
+	// last stores the time of the last update.
+	last time.Time
+
+	// start stores when the service was started.
+	start time.Time
+	// aged is the time if no updates have happened indicates the service is no longer running.
+	// set to 0 if no age check needed
```

Comment:
`time.Duration` zero value is 0 (I assume).
So could just say `// Default zero value means no age check required.`